tools/offline_log_viewer/controller.py

```diff
@@ -820,6 +820,49 @@ def decode_user_and_credential(rdr: Reader):
         })
 
 
+def decode_unresolved_address_serde(rdr: Reader, version: int):
+    decoded = {
+        'host': rdr.read_string(),
+        'port': rdr.read_uint16(),
+        'family': rdr.read_optional(Reader.read_uint16)
```

Comment:
does python guarantee that these reads will be done in order?

Reply:
seems so https://docs.python.org/3/reference/expressions.html#dictionary-displays

Reply:
Ahh cool